### ISSUE TITLE
Add -Wtype-limits to -Wextra for GCC compatibility

### ISSUE THREAD

**Issue by AaronBallman** [llvm/llvm-project#58375]
GCC added the -Wtype-limits warning group to -Wextra around GCC 4.4 and the group has some very helpful extra warnings like tautological comparison type limit warnings (comparing an unsigned int to see if it's positive, etc).

https://godbolt.org/z/nvj16WE71

**Comment by JoeLoser** [comment]
I'm in favor of this.  @nico have your opinions changed on the matter of matching GCC with `-Wextra` in some things like this (based on your comment from https://reviews.llvm.org/D51545)?

**Comment by dwblaikie** [comment]
Yeah, +1, seems reasonable that we might want some rough matching with GCC's `-Wextra`. (I could see an argument for being less consistent with `-Wall`, but even then, consistency with GCC seems plausible)

Rather than going off and commenting on/necromancing D51545, figure we might as well hash that out here - but maybe revisit D51545 once we do (or maybe this issue already subsumes that anyway).

**Comment by AaronBallman** [comment]
I think this request subsumes D51545, so there should be no need to resurrect that review. That one was looking to add just the tautological constant comparison warnings to `-Wextra` but those were added into `-Wtype-limits` so adding `-Wtype-limits` to `-Wextra` will get those diagnostics (and a few more).

**Comment by davidbolvansky** [comment]
Looks like it was reverted

https://reviews.llvm.org/rG897e345042cdf2b3c1dffd234d1b593205f69a32

**Comment by xgupta** [comment]
Yes, I will take a look into the revert this weekend.
